main/test/ca/uwaterloo/flix/library/TestRedBlackTree.flix

```diff
@@ -189,33 +189,81 @@ namespace TestRedBlackTree {
     }
 
 
+    /////////////////////////////////////////////////////////////////////////////
+    // mapWithKey                                                              //
+    /////////////////////////////////////////////////////////////////////////////
+
+    @test
+    def mapWithKey01(): Bool =
+        let Map(t) = Map#{}: Map[Unit, Int32];
+        Map(RedBlackTree.mapWithKey((k, v) -> k + v, t)) == Map#{}
+
+    @test
+    def mapWithKey02(): Bool =
+        let Map(t) = Map#{1 -> 4};
+        Map(RedBlackTree.mapWithKey((k, v) -> k + v, t)) == Map#{1 -> 5}
+
+    @test
+    def mapWithKey03(): Bool =
+        let Map(t) = Map#{2 -> -1, 0 -> 0};
+        Map(RedBlackTree.mapWithKey((k, v) -> k + v, t)) == Map#{2 -> 1, 0 -> 0}
+
+    @test
+    def mapWithKey04(): Bool =
+        let Map(t) = Map#{2 -> -1, 5 -> 15, 11 -> -9};
+        Map(RedBlackTree.mapWithKey((k, v) -> k + v, t)) == Map#{2 -> 1, 5 -> 20, 11 -> 2}
+
+    @test
+    def mapWithKey05(): Bool =
+        let Map(t) = Map#{2 -> -1, 5 -> 15, 11 -> -9, 8 -> 8};
+        Map(RedBlackTree.mapWithKey((k, v) -> k + v, t)) == Map#{2 -> 1, 5 -> 20, 11 -> 2, 8 -> 16}
+
+
     /////////////////////////////////////////////////////////////////////////////
     // map                                                                     //
     /////////////////////////////////////////////////////////////////////////////
 
     @test
     def map01(): Bool =
         let Map(t) = Map#{}: Map[Unit, Int32];
-        Map(RedBlackTree.mapWithKey((_, v) -> v, t)) == Map#{}
+        Map(RedBlackTree.map(v -> v, t)) == Map#{}
 
     @test
     def map02(): Bool =
         let Map(t) = Map#{1 -> 4};
-        Map(RedBlackTree.mapWithKey((k, v) -> k + v, t)) == Map#{1 -> 5}
+        Map(RedBlackTree.map(v -> v + 1, t)) == Map#{1 -> 5}
 
     @test
     def map03(): Bool =
         let Map(t) = Map#{2 -> -1, 0 -> 0};
-        Map(RedBlackTree.mapWithKey((k, v) -> k + v, t)) == Map#{2 -> 1, 0 -> 0}
+        Map(RedBlackTree.map(v -> v + 1, t)) == Map#{2 -> 0, 0 -> 1}
 
     @test
     def map04(): Bool =
         let Map(t) = Map#{2 -> -1, 5 -> 15, 11 -> -9};
-        Map(RedBlackTree.mapWithKey((k, v) -> k + v, t)) == Map#{2 -> 1, 5 -> 20, 11 -> 2}
+        Map(RedBlackTree.map(v -> v + 1, t)) == Map#{2 -> 0, 5 -> 16, 11 -> -8}
 
     @test
     def map05(): Bool =
         let Map(t) = Map#{2 -> -1, 5 -> 15, 11 -> -9, 8 -> 8};
-        Map(RedBlackTree.mapWithKey((k, v) -> k + v, t)) == Map#{2 -> 1, 5 -> 20, 11 -> 2, 8 -> 16}
+        Map(RedBlackTree.map(v -> v + 1, t)) == Map#{2 -> 0, 5 -> 16, 11 -> -8, 8 -> 9}
+
```

Comment:
I would like 5 tests that test `parMap` and `seqMap` directly in this style. (Note this avoid the problem with having a minimum size). Nevertheless I would aim for a tree that contains 1_000 elements. We don't want the tests to be too slow. 

Note that you can also use the trick of observing the order of effects (similarly to LazyList). That might be even more powerful.

Reply:
I'll add effect observing tests in an upcoming commit.